frontend/src/components/QueryBox.js

```diff
@@ -70,11 +71,17 @@ class QueryBox extends Component {
     };
 
     let meta = '';
-    if (resultMeta) {
+    if (result && result.response && result.response.meta) {
       meta = (
-        <span className="meta">
+        <span className="meta meta-success">
           <SuccessIcon className="big-icon" />Showing rows (query took{' '}
-          {resultMeta.elapsedTime / 1000} seconds)
+          {result.response.meta.elapsedTime / 1000} seconds)
+        </span>
+      );
+    } else if (result && result.errorMsg) {
```

Comment:
could you please move it `meta` to functional component?
it would simplify code. something like:
```
if (!result) {
  return null;
}
if (result.errorMsg) {
  return ...
}
...
```

Reply:
Done in e549534